src/netlink.c

```diff
@@ -230,8 +230,29 @@ netlink_link_add_bridge(struct netlink_handle *const hndl,
 	mnl_attr_put_str(nlh, IFLA_IFNAME, name);
 	link_attr = mnl_attr_nest_start(nlh, IFLA_LINKINFO);
 	mnl_attr_put_str(nlh, IFLA_INFO_KIND, "bridge");
-	mnl_attr_nest_end(nlh, link_attr);
 
+	/* Disable multicast snooping on the bridge.
+	 * Netlink support for this was added in kernel 4.4 where
+	 * IFLA_BR_MCAST_SNOOPING was defined with value 23 in an enum.
+	 * Reference:
+	 * https://github.com/torvalds/linux/blob/master/include/uapi/linux/if_link.h
+	 *
+	 * So check if the last value of the enum is greater than 23, before
+	 * adding the value to the netlink nested attributes to be compatible
+	 * with earlier kernels.
+	 */
+	#ifdef IFLA_BR_MAX
```

Comment:
I'm afraid that may not work. One could have a recent enough kernel and the snooping code not compiled in. Probably need to double check that but, conceptually, the kernel may/should reject that value if the code to handle it isn't loaded/compiled in. 

A couple of things you may find interesting:
- you don't need that ifdef, the preprocessor will evaluate it to 0 if undefined.
- should it have been needed, there's a more elegant construct that can be nice to use in the future, eg.
```
#if defined(FOO) && FOO > 23
```